tests/types/test_lazy_types.py

```diff
@@ -195,3 +196,52 @@ def test_lazy_function_in_union():
     [type1, type2] = resolved.types
     assert type1.resolve_type() is LaziestType
     assert type2.resolve_type() is LazyEnum
+
+
+def test_optional_lazy_type_using_or_operator():
```

Comment:
This was already working fine because this file doesn't import future annotations